### PR TITLE
Dispatch 1948

### PR DESCRIPTION
**Pull request by kgiusti**

### CODE REVIEW

**Review by ganeshmurthy — APPROVED** [review]

File: src/alloc_pool.c [hunk 1]
```diff
@@ -435,7 +437,8 @@ void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p)
     assert (desc->header  == PATTERN_FRONT);
     assert (desc->trailer == PATTERN_BACK);
     assert (item->header  == PATTERN_FRONT);
-    assert (*((uint32_t*) (p + desc->total_size)) == PATTERN_BACK);
+    const uint32_t pb = PATTERN_BACK;
```
Comment (jiridanek) [inline src/alloc_pool.c:440]:
```
/home/travis/build/apache/qpid-dispatch/src/alloc_pool.c:440:20: error: unused variable 'pb' [-Werror,-Wunused-variable]

    const uint32_t pb = PATTERN_BACK;
```

Comment (jiridanek) [reply src/alloc_pool.c:440]: (That's because the next line is an assert, so it may not be always compiled in...)

File: src/alloc_pool.c [hunk 2]
```diff
@@ -435,7 +437,8 @@ void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p)
...
+    assert (memcmp(p + desc->total_size, &pb, sizeof(pb)) == 0);
```
Comment (jiridanek) [inline src/alloc_pool.c:441]: I imagine Coverity will scream about this. Memcpy has side-efects and it is being used in an assert.

**Review by jiridanek — CHANGES_REQUESTED** [review]
Looks like it fixed the ubsan warnings, except this one kind

```
1: ../src/parse.c:509:74: runtime error: left shift of 128 by 24 places cannot be represented in type 'int'
```

File: src/alloc_pool.c (hunk 2 above)
Comment (jiridanek) [reply src/alloc_pool.c:441]: Uh, memcmp, never mind.

**Review by jiridanek — APPROVED** [review]
lgtm.

tests failed on GHA with known issue

```
 69: qdrouterd: ../src/router_core/transfer.c:225: qdr_link_process_deliveries: Assertion `(link->undelivered).head' failed.
```

https://github.com/apache/qpid-dispatch/pull/1015/checks?check_run_id=1826269317#step:9:2078

And on Travis with some (old) TSAN warnings

**Pull request closed without merging**